client/src/modules/list/components/Items/ItemsList.jsx

```diff
@@ -12,41 +12,45 @@ import MessageBox from 'common/components/MessageBox';
 import { MessageType } from 'common/constants/enums';
 import { disableItemAnimations } from './model/actions';
 import { RouterMatchPropType } from 'common/constants/propTypes';
-import { updateLimit } from 'modules/list/model/actions';
 import { DISPLAY_LIMIT } from 'common/constants/variables';
+import { updateLimit } from 'modules/list/model/actions';
 
 class ItemsList extends PureComponent {
   constructor(props) {
     super(props);
 
     this.state = {
+      displayedItems: null,
       limit: DISPLAY_LIMIT
     };
   }
 
-  showMore = () =>
-    this.setState(
-      ({ limit }) => ({ limit: limit + DISPLAY_LIMIT }),
-      this.handleLimitUpdate
-    );
+  componentDidMount() {
+    this.prepareDisplayedItems();
+  }
 
-  showLess = () =>
-    this.setState({ limit: DISPLAY_LIMIT }, this.handleLimitUpdate);
+  componentDidUpdate(previousProps) {
+    const { items: previousItems } = previousProps;
+    const { items } = this.props;
 
-  handleDisableAnimations = item => () => {
-    const { _id: itemId, animate } = item;
-    if (animate) {
-      const {
-        disableItemAnimations,
-        match: {
-          params: { id: listId }
-        }
-      } = this.props;
-
-      disableItemAnimations(itemId, listId);
+    if (previousItems.length !== items.length) {
+      this.prepareDisplayedItems();
     }
+  }
+
+  prepareDisplayedItems = () => {
+    const { limit } = this.state;
+    const { items } = this.props;
+    const displayedItems = items.slice(0, limit).map(item => item);
```

Comment:
Consider rename `displayed` to `display` - if you apply this, please make sure to rename methods as well (`prepareDisplayItems` etc.).